### PR TITLE
Move function bodies (non methods) to cpp.

### PR DESCRIPTION
**Pull request by jean-m-cyr**
- These functions were inlined for convenience, to avoid
  multiple declaration errors. This is not a good reason to
  inline a function.
- None of these are used in performance critical areas

### CODE REVIEW

File: libethcore/Miner.cpp [hunk 1]
```diff
@@ -1,19 +1,101 @@
+/*
+ This file is part of cpp-ethereum.
```
Comment (chfast) [inline libethcore/Miner.cpp]: I'm not sure this is accurate :)

File: libethcore/Miner.h [hunk 2]
```diff
@@ -79,36 +79,17 @@ struct HwMonitor
...
-}
+std::ostream& operator<<(std::ostream& os, HwMonitor _hw);
 
-class FormatMemSize
+class FormattedMemSize
 {
 public:
-    FormatMemSize() = delete;
-    FormatMemSize(uint64_t s) { m_size = s; }
+    FormattedMemSize() = delete;
```
Comment (chfast) [inline libethcore/Miner.h]: This is not needed if you declared any other constructor.

File: libethcore/Miner.h [hunk 3]
```diff
@@ -79,36 +79,17 @@ struct HwMonitor
...
+    FormattedMemSize(uint64_t s) { m_size = s; }
```
Comment (chfast) [inline libethcore/Miner.h]: This should be `explicit` and `noexcept`.

File: libethcore/Miner.cpp (hunk 1 above)
Comment (jean-m-cyr) [reply libethcore/Miner.cpp]: It isn't. I'll fix.

File: libethcore/Miner.h (hunk 2 above)
Comment (jean-m-cyr) [reply libethcore/Miner.h]: No default constructor if any other constructor is declared? I did not know that!

File: libethcore/Miner.h (hunk 3 above)
Comment (jean-m-cyr) [reply libethcore/Miner.h]: Ok

File: libethcore/Miner.cpp (hunk 1 above)
Comment (chfast) [inline libethcore/Miner.cpp]: Almost all.

File: libethcore/Miner.cpp [hunk 4]
```diff
@@ -1,19 +1,101 @@
...
+    os << _hw.tempC << "C " << _hw.fanP << "%";
+    if (_hw.powerW)
+        os << ' ' << fixed << setprecision(0) << _hw.powerW << "W";
+    return os;
+}
+
+std::ostream& operator<<(std::ostream& os, FormattedMemSize s)
+{
+    static const char* suffixes[] = {"bytes", "KB", "MB", "GB"};
+    double d = s.m_size;
```
Comment (XhmikosR) [inline libethcore/Miner.cpp]: @jean-m-cyr: trivial warning on Windows here

```
C:\Users\xmr\Desktop\ethminer\libethcore\Miner.cpp(49): warning C4244: 'initializing': conversion from 'uint64_t' to
'double', possible loss of data [C:\Users\xmr\Desktop\ethminer\build\libethcore\ethcore.vcxproj]
```

Comment (jean-m-cyr) [reply libethcore/Miner.cpp]: Easy fix: double d = double(s.m_size);

Submit a PR if you want.

**Pull request merged**